File: code/classes/pattern-matching.ex
defmodule PatternMatching do
  def describe(1),                    do: "One"
  def describe("Hello"),              do: "Greeting"
  def describe(x) when is_float(x),   do: "Float"
  def describe(x) when is_integer(x), do: "Integer"
  def describe(x) when is_binary(x),  do: "String"
  def describe(_),                    do: "Unknown"
end

IO.puts PatternMatching.describe(1)
IO.puts PatternMatching.describe("Hello")
IO.puts PatternMatching.describe(3.14)

File: code/classes/pattern-matching.erl
-module(pattern_matching).
-export([describe/1, start/0]).

describe(1)                -&gt; "One";
describe(hello)            -&gt; "Greeting";
describe(X) when is_float(X) -&gt; "Float";
describe(X) when is_integer(X) -&gt; "Integer";
describe(X) when is_list(X)    -&gt; "String";
describe(_)                -&gt; "Unknown".

start() -&gt;
    io:format("~s~n", [describe(1)]),
    io:format("~s~n", [describe(hello)]),
    io:format("~s~n", [describe(3.14)]).
